clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -494,10 +572,32 @@ private void handleDisconnections(List<ClientResponse> responses, long now) {
     /**
      * Record any newly completed connections
      */
-    private void handleConnections() {
+    private void handleConnections(long now) {
         for (String node : this.selector.connected()) {
             log.debug("Completed connection to node {}", node);
             this.connectionStates.connected(node);
+            // Though the node is connected, we might not still be able to send requests. For instance,
+            // in case of SSL connection, SSL handshake happens after connection is established.
+            if (!apiVersionsRequestQueue.contains(node))
+                apiVersionsRequestQueue.add(node);
+        }
+        maybeInitiateApiVersionsFetch(now);
+    }
+
+    private void maybeInitiateApiVersionsFetch(long now) {
+        if (this.requiredApiVersions == null)
+            return;
+
+        for (String node : apiVersionsRequestQueue) {
+            if (selector.isChannelReady(node) && inFlightRequests.canSendMore(node)) {
```

Comment:
`canSendRequest` also checks if connection state is `Ready`, which won't be true here.